nft_ingester/src/bin/backfill/main.rs

```diff
@@ -197,82 +198,85 @@ async fn main() {
+                    };
+                    progress_bar.set_message(format!(
+                        "Slots Processed: {} Current Slot: {} Rate: {:.2}/s",
+                        current_slots_processed, slot, current_rate
+                    ));
                 }
-
-                // Increment slots_processed
-                let current_slots_processed = slots_processed.fetch_add(1, Ordering::Relaxed) + 1;
-
-                // Update progress bar position and message
-                progress_bar.inc(1);
-
-                let current_rate = {
-                    let rate_guard = rate.lock().unwrap();
-                    *rate_guard
-                };
-                progress_bar.set_message(format!(
-                    "Slots Processed: {} Current Slot: {} Rate: {:.2}/s",
-                    current_slots_processed, slot, current_rate
-                ));
             }
         });
 
         worker_handles.push(handle);
     }
 
-    // Spawn a task to update the rate periodically
-    let slots_processed_clone = slots_processed.clone();
-    let rate_clone = rate.clone();
-    let shutdown_token_clone = shutdown_token.clone();
+    usecase::executor::spawn({
+        let cancellation_token = cancellation_token.clone();
+        let slots_processed = slots_processed.clone();
+        let rate = rate.clone();
+        async move {
+            let mut last_time = std::time::Instant::now();
+            let mut last_count = slots_processed.load(Ordering::Relaxed);
 
-    tokio::spawn(async move {
-        let mut last_time = std::time::Instant::now();
-        let mut last_count = slots_processed_clone.load(Ordering::Relaxed);
+            loop {
+                tokio::time::sleep(std::time::Duration::from_secs(1)).await;
 
-        loop {
-            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                if cancellation_token.is_cancelled() {
+                    break;
+                }
 
-            if shutdown_token_clone.is_cancelled() {
-                break;
-            }
+                let current_time = std::time::Instant::now();
+                let current_count = slots_processed.load(Ordering::Relaxed);
 
-            let current_time = std::time::Instant::now();
-            let current_count = slots_processed_clone.load(Ordering::Relaxed);
+                let elapsed = current_time.duration_since(last_time).as_secs_f64();
+                let count = current_count - last_count;
 
-            let elapsed = current_time.duration_since(last_time).as_secs_f64();
-            let count = current_count - last_count;
+                let current_rate = if elapsed > 0.0 { (count as f64) / elapsed } else { 0.0 };
 
-            let current_rate = if elapsed > 0.0 { (count as f64) / elapsed } else { 0.0 };
+                // Update rate
+                {
+                    let mut rate_guard = rate.lock().unwrap();
+                    *rate_guard = current_rate;
+                }
```

Comment:
```suggestion
                *rate.lock().unwrap() = current_rate;
```

Isn't it possible? Why do we need a dedicated block for it? I guess it will be freed at the end of the loop, won't it?

Reply:
this change is out of scope for this PR. i will create a task gathering such places and will submit the refactoring changes in a later PR